main.rb

```diff
@@ -0,0 +1,164 @@
+    arr = to_a
+    index = 0
+    i = 0
+    while i < arr.size
+      yield(arr[i], index)
+      index += 1
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    self.class == Hash ? item = {} : item = []
+    my_each { |i| item << i if yield(i) } if self.class != Hash
+    my_each_with_index { |k| item << k[0] if yield(k[0]) } if self.class == Hash
+    item
+  end
+
+  def my_all?(*args)
+    flag = 0
+    self.class == Hash || self.class == Range ? arr = to_a : arr = self
+    case args.empty?
+    when true
+      if block_given?
+        arr.my_each do |i|
+          break if yield(i).nil? || yield(i) == false
+
+          flag += 1 if yield(i) != false || yield(i).nil?
+        end
+      else
+        arr.my_each { |i| flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      if args[0].class == Regexp
+        arr.my_each { |itr| flag += 1 if itr =~ args[0] }
+      elsif args[0].class == Class
+        arr.my_each { |itr| flag += 1 if itr.is_a?(args[0]) }
+      else
+        arr.my_each { |itr| flag += 1 if args[0] == itr }
+      end
+    end
+    flag == size
+  end
+
+  def my_any?(*args)
+    flag = 0
+    case args.empty?
+    when true
+      if block_given?
+        my_each { |i| flag += 1 if yield(i) }
+      else
+        my_each { |i| flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      if args[0].class == Regexp
+        my_each { |itr| flag += 1 if itr =~ args[0] }
+      elsif args[0].class == Class
+        my_each { |itr| flag += 1 if itr.is_a?(args[0]) }
+      else
+        my_each { |itr| flag += 1 if args[0] == itr }
+      end
+    end
+    flag.positive?
+  end
+
+  def my_none?(*args)
+    flag = 0
+    case block_given?
+    when true
+      my_each { |i| flag += 1 if yield(i) == true }
+    when false
+      if args.empty?
+        my_each { |i| flag += 1 if i == true || !i.to_s.empty? }
+      elsif args[0].class == Regexp
+        my_each { |i| flag += 1 if i =~ args[0] }
+      elsif args[0].class == Class
+        my_each { |i| flag += 1 if i.is_a?(args[0]) }
+      else
+        my_each { |i| flag += 1 if i == args[0] }
+      end
+    end
+    flag.zero?
+  end
+
+  def my_count(*arg)
+    counter = 0
+    case !block_given?
+    when true
+      if arg.empty?
+        size
+      else
+        my_each { |i| counter += 1 if arg[0] == i }
+        counter
+      end
+    when false
+      my_each { |i| counter += 1 if yield(i) == true }
+      counter
+    end
+  end
+
+  def my_map(*arg)
+    return to_enum(:my_map) unless block_given?
+
+    new_array = []
+    case block_given?
+    when true && !arg.empty?
+      if arg[0].class == Regexp
+        my_each { |i| yield(i) ? new_array << proc { arg } : nil }
+      else
+        my_each { |i| new_array << i if arg[0].call(i) == true }
+      end
+    when true
+      my_each do |i|
+        new_array << yield(i)
+      end
+    end
+    new_array
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```